Branch-SDK/src/io/branch/referral/RemoteInterface.java

```diff
@@ -104,18 +105,24 @@ private boolean addCommonParams(JSONObject post) {
 		return false;
 	}
 	
-	public ServerResponse make_restful_get(String url, String tag, int timeout, boolean log) {
-		JSONObject post = new JSONObject();
-		if (addCommonParams(post)) {
-			url += this.convertJSONtoString(post);
+	private ServerResponse make_restful_get(String baseUrl, String tag, int timeout, int retryNumber, boolean log) {
+		String modifiedUrl = baseUrl;
+		JSONObject getParameters = new JSONObject();
+		if (addCommonParams(getParameters, retryNumber)) {
+			modifiedUrl += this.convertJSONtoString(getParameters);
 		} else {
 			return new ServerResponse(tag, NO_BRANCH_KEY_STATUS);
 		}
 		
-		try {    	
-			if (log) PrefHelper.Debug("BranchSDK", "getting " + url);
-		    HttpGet request = new HttpGet(url);
+		try {
+			if (log) PrefHelper.Debug("BranchSDK", "getting " + modifiedUrl);
+		    HttpGet request = new HttpGet(modifiedUrl);
 		    HttpResponse response = getGenericHttpClient(timeout).execute(request);
+			if (response.getStatusLine().getStatusCode() >= 500 &&
+				retryNumber < prefHelper_.getRetryCount()) {
+				retryNumber++;
+				make_restful_get(baseUrl, tag, timeout, retryNumber, log);
```

Comment:
When the recursive calls unwind, they leave no change on the response object, isn't it?